lib/Service/ImageService.php

```diff
@@ -0,0 +1,596 @@
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OCA\Text\Service;
+
+use Exception;
+use OCA\Text\Controller\ImageController;
+use OCP\Constants;
+use OCP\Files\Folder;
+use OCP\Files\File;
+use OCP\Files\NotFoundException;
+use OCP\Files\SimpleFS\ISimpleFile;
+use OCP\IPreview;
+use OCP\Share\Exceptions\ShareNotFound;
+use OCP\Share\IShare;
+use Throwable;
+use GuzzleHttp\Exception\ClientException;
+use GuzzleHttp\Exception\ConnectException;
+use GuzzleHttp\Exception\ServerException;
+use OCA\Text\AppInfo\Application;
+use OCP\Http\Client\IClientService;
+use OCP\Files\IRootFolder;
+use Psr\Log\LoggerInterface;
+use OCP\Share\IManager as ShareManager;
+use function json_encode;
+use function preg_replace;
+
+class ImageService {
+
+	/**
+	 * @var ShareManager
+	 */
+	private $shareManager;
+	/**
+	 * @var IRootFolder
+	 */
+	private $rootFolder;
+	/**
+	 * @var IClientService
+	 */
+	private $clientService;
+	/**
+	 * @var LoggerInterface
+	 */
+	private $logger;
+	/**
+	 * @var IPreview
+	 */
+	private $previewManager;
+
+	public function __construct(IRootFolder $rootFolder,
+								LoggerInterface $logger,
+								ShareManager $shareManager,
+								IPreview $previewManager,
+								IClientService $clientService) {
+		$this->rootFolder = $rootFolder;
+		$this->shareManager = $shareManager;
+		$this->clientService = $clientService;
+		$this->logger = $logger;
+		$this->previewManager = $previewManager;
+	}
+
+	/**
+	 * Get image content from file name
+	 *
+	 * @param int $textFileId
+	 * @param string $imageFileName
+	 * @param string $userId
+	 * @return File|null
+	 * @throws NotFoundException
+	 * @throws \OCP\Files\InvalidPathException
+	 * @throws \OCP\Files\NotPermittedException
+	 * @throws \OCP\Lock\LockedException
+	 * @throws \OC\User\NoUserException
+	 */
+	public function getImage(int $textFileId, string $imageFileName, string $userId): ?ISimpleFile {
+		$textFile = $this->getTextFile($textFileId, $userId);
+		$attachmentFolder = $this->getOrCreateAttachmentDirectoryForFile($textFile);
+		if ($attachmentFolder !== null) {
+			try {
+				$imageFile = $attachmentFolder->get($imageFileName);
+			} catch (NotFoundException $e) {
+				return null;
+			}
+			if ($imageFile instanceof File) {
+//				return $imageFile;
```

Comment:
- [ ] To be discussed if we still need the original file for gif support for example